File: 1480.running-sum-of-1-d-array.py
#
# @lc app=leetcode id=1480 lang=python3
#
# [1480] Running Sum of 1d Array
#

# @lc code=start
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        list = []
        for i, v in enumerate(nums):
            list.append(sum(nums[0:i+1]))
        return list
# @lc code=end

File: 1480/1480.running-sum-of-1-d-array.c
/*
 * @lc app=leetcode id=1480 lang=c
 *
 * [1480] Running Sum of 1d Array
 */

// @lc code=start
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* runningSum(int* nums, int numsSize, int* returnSize){
    for (int i = 1; i < numsSize; i++)
    {
        nums[i] += nums[i-1];
    }
    *returnSize = numsSize;
    return nums;
}
// @lc code=end

